src/adcm_client/packer/types.py

```diff
@@ -57,11 +57,32 @@ def _get_top_dirs(image: Image, prepared_image: Image, client: DockerClient) ->
         map(lambda x: x.split('==')[0], set(modified_module_list).difference(default_module_list))
     )
 
-    modules_data = yaml.safe_load_all(
-        client.containers.run(
-            prepared_image, f'pip show -f {" ".join(modules)}', remove=True
-        ).decode("utf-8")
-    )
+    pip_show_output = client.containers.run(
+        prepared_image, f'pip show -f {" ".join(modules)}', remove=True
+    ).decode("utf-8")
+
+    # pip show returns not valid yaml, e.g. for pip show ipython
+    # we can see "Summary: IPython: Productive Interactive Computing"
+    # so we should extract only required fields (Location, Files and children of Files)
+    resulting_lines = []
+    # indicates if we try to parse children of Files section
+    in_files_section = False
+    for line in pip_show_output.splitlines():
+        # yaml doc separator or Location
+        if line == "---" or line.startswith("Location"):
+            resulting_lines.append(line)
+            in_files_section = False
+        # we want to pass Files into results and its children
+        elif line.startswith("Files"):
+            resulting_lines.append(line)
+            in_files_section = True
+        # we are inside children of Files
+        elif in_files_section and line.startswith("  "):
+            resulting_lines.append(line)
+        else:
+            in_files_section = False
```

Comment:
I think this should be a separate function like "sanitize input"  with doctests/tests for the function itself

Now it's very complicated to read and understand what's the expected output

With separate function we'll see if it can be reworked somehow based on input and output presented in tests

Reply:
Good point, thank you! Let me split it and prepare unit test.

Reply:
Modified.
Few comments:

1. I am not sure that unit tests running in CI
2. I can see strange error `mkdir: can't create directory '/tmp/build_with_requirements_83tq3m3l/community/pmod': No such file or directory\ncp:...': No such file or directory\nchown: /tmp/build_with_requirements_83tq3m3l/community/pmod: No such file or directory\n"`. Looks like it come during creation of pmod directory, which looks not related with PR changes.